conftest.py

```diff
@@ -37,14 +38,19 @@ def pytest_collection_modifyitems(config, items):
 
     for item in items:
         rel_path = Path(item.fspath).relative_to(rootdir)
-
-        if "si_based_sorters" not in rel_path.name:
+        if "sorters" not in str(rel_path):
             for mark_name in mark_names:
                 if f"/{mark_name}/" in str(rel_path):
                     mark = getattr(pytest.mark, mark_name)
                     item.add_marker(mark)
         else:
-            item.add_marker("si_sorters")
+            if "/internal/" in str(rel_path):
+                item.add_marker("sorters_internal")
+            elif "/external/" in str(rel_path):
+                item.add_marker("sorters_external")
+            else:
+                print(rel_path)
```

Comment:
I guess this print statement should be removed?

It would be nice if we could already do use https://peps.python.org/pep-0634/ but I guess we want people who use older versions to still be able to run the tests?

Reply:
I would also do the if with the positive side, I think that:
```
        rel_path = Path(item.fspath).relative_to(rootdir)
        if "sorters" in str(rel_path):
            if "/internal/" in str(rel_path):
                item.add_marker("sorters_internal")
            elif "/external/" in str(rel_path):
                item.add_marker("sorters_external")
            else:
                item.add_marker("sorters")
        else:
            for mark_name in mark_names:
                if f"/{mark_name}/" in str(rel_path):
                    mark = getattr(pytest.mark, mark_name)
                    item.add_marker(mark)

```

Reply:
done